pkg/providers/imagefamily/customscriptsbootstrap/utils_test.go

```diff
@@ -208,3 +211,202 @@ func TestHydrateBootstrapTokenIfNeeded(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertLocalDNSToModel(t *testing.T) {
+	tests := []struct {
+		name     string
+		localDNS *v1beta1.LocalDNS
+		expected *models.LocalDNSProfile
+	}{
+		{
+			name:     "Nil LocalDNS",
+			localDNS: nil,
+			expected: nil,
+		},
+		{
+			name: "LocalDNS with mode only",
+			localDNS: &v1beta1.LocalDNS{
+				Mode: lo.ToPtr(v1beta1.LocalDNSModeRequired),
+			},
+			expected: &models.LocalDNSProfile{
+				Mode: lo.ToPtr("Required"),
+			},
+		},
+		{
+			name: "LocalDNS with VnetDNSOverrides",
+			localDNS: &v1beta1.LocalDNS{
+				Mode: lo.ToPtr(v1beta1.LocalDNSModePreferred),
+				VnetDNSOverrides: map[string]*v1beta1.LocalDNSOverrides{
+					"example.com": {
+						QueryLogging: lo.ToPtr(v1beta1.LocalDNSQueryLoggingLog),
+						Protocol:     lo.ToPtr(v1beta1.LocalDNSProtocolForceTCP),
+					},
+				},
+			},
+			expected: &models.LocalDNSProfile{
+				Mode: lo.ToPtr("Preferred"),
+				VnetDNSOverrides: models.LocalDNSOverrides{
+					"example.com": models.LocalDNSOverride{
+						QueryLogging: lo.ToPtr("Log"),
+						Protocol:     lo.ToPtr("ForceTCP"),
+					},
+				},
+			},
+		},
+		{
+			name: "LocalDNS with KubeDNSOverrides",
+			localDNS: &v1beta1.LocalDNS{
+				KubeDNSOverrides: map[string]*v1beta1.LocalDNSOverrides{
+					"cluster.local": {
+						ForwardDestination: lo.ToPtr(v1beta1.LocalDNSForwardDestinationClusterCoreDNS),
+						MaxConcurrent:      lo.ToPtr(int32(100)),
+					},
+				},
+			},
+			expected: &models.LocalDNSProfile{
+				KubeDNSOverrides: models.LocalDNSOverrides{
+					"cluster.local": models.LocalDNSOverride{
+						ForwardDestination: lo.ToPtr("ClusterCoreDNS"),
+						MaxConcurrent:      lo.ToPtr(int32(100)),
+					},
+				},
+			},
+		},
+		{
+			name: "LocalDNS with both VnetDNS and KubeDNS overrides",
+			localDNS: &v1beta1.LocalDNS{
+				Mode: lo.ToPtr(v1beta1.LocalDNSModeDisabled),
+				VnetDNSOverrides: map[string]*v1beta1.LocalDNSOverrides{
+					"vnet.domain": {
+						Protocol: lo.ToPtr(v1beta1.LocalDNSProtocolPreferUDP),
+					},
+				},
+				KubeDNSOverrides: map[string]*v1beta1.LocalDNSOverrides{
+					"kube.domain": {
+						ForwardPolicy: lo.ToPtr(v1beta1.LocalDNSForwardPolicyRoundRobin),
+					},
+				},
+			},
+			expected: &models.LocalDNSProfile{
+				Mode: lo.ToPtr("Disabled"),
+				VnetDNSOverrides: models.LocalDNSOverrides{
+					"vnet.domain": models.LocalDNSOverride{
+						Protocol: lo.ToPtr("PreferUDP"),
+					},
+				},
+				KubeDNSOverrides: models.LocalDNSOverrides{
+					"kube.domain": models.LocalDNSOverride{
+						ForwardPolicy: lo.ToPtr("RoundRobin"),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := convertLocalDNSToModel(tt.localDNS)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestConvertLocalDNSOverrideToModel(t *testing.T) {
```

Comment:
minor, style: Consider just putting these tests into the above test and passing everything through `convertLocalDNSToModel` (test the public interface or at least the most public part)